core/src/main/java/com/linecorp/armeria/server/healthcheck/DefaultMemoryHealthChecker.java

```diff
@@ -0,0 +1,51 @@
+package com.linecorp.armeria.server.healthcheck;
+
+import java.lang.management.BufferPoolMXBean;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryPoolMXBean;
+import java.lang.management.MemoryType;
+import java.lang.management.MemoryUsage;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.google.common.base.MoreObjects;
+
+public class DefaultMemoryHealthChecker implements HealthChecker {
```

Comment:
Could be package private final:
```suggestion
final class DefaultMemoryHealthChecker implements HealthChecker {
```